packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -1,43 +1,122 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { store as blocksStore } from '@wordpress/blocks';
+import { privateApis as blocksPrivateApis } from '@wordpress/blocks';
 import {
-	BaseControl,
-	PanelBody,
-	__experimentalHStack as HStack,
 	__experimentalItemGroup as ItemGroup,
 	__experimentalItem as Item,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+	__experimentalTruncate as Truncate,
+	__experimentalDropdownContentWrapper as DropdownContentWrapper,
+	Dropdown,
 } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch, useRegistry } from '@wordpress/data';
+import { useContext } from '@wordpress/element';
+import { useViewportMatch } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
-import { canBindAttribute } from '../hooks/use-bindings-attributes';
+import {
+	canBindAttribute,
+	getBindableAttributes,
+} from '../hooks/use-bindings-attributes';
+import { store as blockEditorStore } from '../store';
 import { unlock } from '../lock-unlock';
 import InspectorControls from '../components/inspector-controls';
+import BlockContext from '../components/block-context';
+
+const popoverProps = {
+	placement: 'left-start',
+	offset: 36,
+	shift: true,
+};
+
+const useToolsPanelDropdownMenuProps = () => {
+	const isMobile = useViewportMatch( 'medium', '<' );
+	return ! isMobile
+		? {
+				popoverProps: {
+					placement: 'left-start',
+					// For non-mobile, inner sidebar width (248px) - button width (24px) - border (1px) + padding (16px) + spacing (20px)
+					offset: 259,
+				},
+		  }
+		: {};
+};
```

Comment:
I agree with that, but, at the same time, we would need to allow this option to be extensible and editable for external developers.

Reply:
Pinging @WordPress/gutenberg-components because they will probably have more context if this makes sense or feasible.

Reply:
Thank you for the ping!

Those popover settings (placement and offset) are related to the block inspector sidebar. The `@wordpress/components` package needs to provide UI components to more consumers than just the block editor, and therefore it should not be aware of those details otherwise it would become tightly coupled with the editor.

I agree though that it would be best to store those settings in a common place.

Since those settings are related to the sidebar, I would suggest storing them in a package that is a common dependency of the block editor, but not a generic package (like components, data, compose, gets). 

Maybe in the `interface` package (where complementary area, the component used for the sidebar, [is defined](https://github.com/WordPress/gutenberg/blob/61659a34f7af204c20860b207a04963439ae6842/packages/interface/src/components/complementary-area/index.js#L45)), or even in the `@wordpress/editor` package itself?

Reply:
Okay, thanks for the context! We can figure out the best place for this in a follow-up pull request I believe.

Reply:
> Maybe in the @wordpress/interface package (where complementary area, the component used for the sidebar, [is defined](https://github.com/WordPress/gutenberg/blob/61659a34f7af204c20860b207a04963439ae6842/packages/interface/src/components/complementary-area/index.js#L45)), or even in the @wordpress/editor package itself?

To my knowledge, there is intent to sunset `@wordpress/interface` in favor of `@wordpress/editor`.

Reply:
I have created [this follow-up pull request](https://github.com/WordPress/gutenberg/pull/64105) to use only one util and keep discussing it.